src/Umbraco.Infrastructure/Persistence/Repositories/Implement/KeyValueRepository.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NPoco;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Migrations;
+using Umbraco.Core.Models;
+using Umbraco.Core.Persistence.Dtos;
+using Umbraco.Core.Persistence.Querying;
+using Umbraco.Core.Scoping;
+using Umbraco.Infrastructure.Persistence.Repositories;
+
+namespace Umbraco.Core.Persistence.Repositories.Implement
+{
+    internal class KeyValueRepository : NPocoRepositoryBase<string, IKeyValue>, IKeyValueRepository
+    {
+        public KeyValueRepository(IScopeAccessor scopeAccessor, ILogger logger)
+            : base(scopeAccessor, AppCaches.NoCache, logger)
+        { }
+
+        public void Initialize()
```

Comment:
I'm not so sure about this one, i don't really think that `Initialize` makes sense on this interface since it's not really clear what it actually does. Since this is specific to migrations i think this code needs to live outside of this repository and the call made elsewhere.

Reply:
Have moved into a new component: `IKeyValueServiceInitialization` (and implementation).